kalite/packages/bundled/securesync/api_client.py

```diff
@@ -13,9 +13,17 @@
 
 class BaseClient(object):
 
-    def __init__(self, host="%s://%s/" % (settings.SECURESYNC_PROTOCOL, settings.CENTRAL_SERVER_HOST), require_trusted=True, verbose=True):
-        self.parsed_url = urllib2.urlparse.urlparse(host)
-        self.url = "%s://%s" % (self.parsed_url.scheme, self.parsed_url.netloc)
+    def __init__(self, host=None, require_trusted=True, verbose=True):
+        
+        if not host:
+            if settings.CENTRAL_SERVER_URL:
+                self.url = settings.CENTRAL_SERVER_URL
+            else:
+                self.url = "%s://%s/" % (settings.SECURESYNC_PROTOCOL, settings.CENTRAL_SERVER_HOST)
+        else:
+            parsed_url = urllib2.urlparse.urlparse(host)
+            self.url = "%s://%s" % (self.parsed_url.scheme, self.parsed_url.netloc)
+        
```

Comment:
I'm not sure why the latter used to call `urllib2.urlparse.urlparse`, it seems that somehow it checks if the `host` is a valid URL with a scheme even though the kwarg was called `host`.. I don't quite grasp this. The `central-develop` branch had this, which would cause a regression on 0.17.x:

```
    def __init__(self, host=None, require_trusted=True, verbose=True):
        self.url = host or settings.CENTRAL_SERVER_URL
        self.parsed_url = urllib2.urlparse.urlparse(self.url)
```